graph-builder/src/registry.rs

```diff
@@ -63,21 +157,21 @@ struct Tags {
     tags: Vec<String>,
 }
 
-fn fetch_tags(registry: &str, repo: &str) -> Result<Vec<String>, Error> {
-    let base = Url::parse(registry)?;
-    let tags: Tags = {
-        let mut response = reqwest::get(base.join(&format!("v2/{}/tags/list", repo))?)
-            .context("failed to fetch image tags")?;
-        ensure!(
-            response.status().is_success(),
-            "failed to fetch image tags: {}",
-            response.status()
-        );
-
-        serde_json::from_str(&response.text()?)?
-    };
+fn fetch_tags(
+    tcore: &mut Core,
```

Comment:
No need to evaluate the future here, just return it. You won't need to pass in the core.